src/pylibsshext/session.pyx

```diff
@@ -302,6 +320,54 @@ cdef class Session(object):
                                       self._fingerprint_py,
                                       know_host_msg)
 
+    def authenticate_specific_pubkey(
+            self,
+            bytes private_key_b64 not None,
+            private_key_password=None,
+    ):
+        """Authenticate this session using a private key.
+
+        If a password is provided, it'll be used to decrypt the key.
+
+        :param private_key_b64: A private key.
+        :type private_key_b64: bytes
+
+        :param private_key_password: A password for the private key \
+                                     (if it's protected), defaults to \
+                                     no password.
+        :type private_key_password: bytes, optional
+
+        :raises LibsshSessionException: If authentication failed.
+
+        :return: Nothing.
+        :rtype: NoneType
+        """
+        cdef const char *c_private_key_b64 = private_key_b64
+        cdef bytes b_password
+        cdef const char *c_password = NULL
+        cdef libssh.ssh_key _private_key
+        cdef int rc
+        if private_key_password is not None:
+            if isinstance(private_key_password, bytes):
+                b_password = private_key_password
+            else:
+                b_password = private_key_password.encode()
+            c_password = b_password
+        libssh.ssh_pki_import_privkey_base64(
+            c_private_key_b64, c_password,
+            NULL, NULL,
+            &_private_key,
```

Comment:
this variable is dynamic memory allocated and should be freed before existing the function